src/ameba/ast/visitors/scope_visitor.cr

```diff
@@ -140,12 +155,18 @@ module Ameba::AST
 
     # :nodoc:
     def visit(node : Crystal::Call)
-      return true unless node.name == "super" && node.args.empty?
-      return true unless (scope = @current_scope).def?
-      scope.arguments.each do |arg|
-        variable = arg.variable
-        variable.reference(variable.node, scope).explicit = false
+      case @current_scope
+      when .def?
+        if node.name == SUPER && node.args.empty?
+          @current_scope.arguments.each do |arg|
+            variable = arg.variable
+            variable.reference(variable.node, @current_scope).explicit = false
+          end
+        end
+      when .top_level?
+        return false if SPECIAL_TOP_LEVEL_MACROS.includes?(node.name)
```

Comment:
For higher precision matching the signature of the `record` macro, you could check if the first argument is a `Crystal::Path`. This obviously requires more customization than just a list of special macro names.